nemo/collections/asr/models/ctc_bpe_models.py

```diff
@@ -0,0 +1,202 @@
+
+    def transcribe(self, path2audio_file: str) -> str:
+        pass
+
+    def _setup_dataloader_from_config(self, config: Optional[Dict]):
+        if 'augmentor' in config:
+            augmentor = process_augmentations(config['augmentor'])
+        else:
+            augmentor = None
+
+        featurizer = WaveformFeaturizer(
+            sample_rate=config['sample_rate'], int_values=config.get('int_values', False), augmentor=augmentor
+        )
+
+        dataset = AudioToBPEDataset(
+            manifest_filepath=config['manifest_filepath'],
+            tokenizer=self.tokenizer,
+            featurizer=featurizer,
+            max_duration=config.get('max_duration', None),
+            min_duration=config.get('min_duration', None),
+            max_utts=config.get('max_utts', 0),
+            trim=config.get('trim_silence', True),
+            load_audio=config.get('load_audio', True),
+            add_misc=config.get('add_misc', False),
+        )
+
+        return torch.utils.data.DataLoader(
+            dataset=dataset,
+            batch_size=config['batch_size'],
+            collate_fn=dataset.collate_fn,
+            drop_last=config.get('drop_last', False),
+            shuffle=config['shuffle'],
+            num_workers=config.get('num_workers', 0),
+        )
+
+    @property
+    def input_types(self) -> Optional[Dict[str, NeuralType]]:
+        if hasattr(self.preprocessor, '_sample_rate'):
+            audio_eltype = AudioSignal(freq=self.preprocessor._sample_rate)
+        else:
+            audio_eltype = AudioSignal()
+        return {
+            "input_signal": NeuralType(('B', 'T'), audio_eltype),
+            "input_signal_length": NeuralType(tuple('B'), LengthsType()),
+        }
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        return {
+            "outputs": NeuralType(('B', 'T', 'D'), LogprobsType()),
+            "encoded_lengths": NeuralType(tuple('B'), LengthsType()),
+            "greedy_predictions": NeuralType(('B', 'T'), LabelsType()),
+        }
+
+    # PTL-specific methods
+    def training_step(self, batch, batch_nb):
+        self.train()
+        audio_signal, audio_signal_len, transcript, transcript_len = batch
+        log_probs, encoded_len, predictions = self.forward(
+            input_signal=audio_signal, input_signal_length=audio_signal_len
+        )
+        loss_value = self.loss(
+            log_probs=log_probs, targets=transcript, input_lengths=encoded_len, target_lengths=transcript_len
+        )
+        wer_num, wer_denom = self._wer(predictions, transcript, transcript_len)
+        tensorboard_logs = {
+            'train_loss': loss_value,
+            'training_batch_wer': wer_num / wer_denom,
+            'learning_rate': self._optimizer.param_groups[0]['lr'],
+        }
+        return {'loss': loss_value, 'log': tensorboard_logs}
+
+    def validation_step(self, batch, batch_idx):
+        self.eval()
```

Comment:
Doesn't PTL do this automatically? Should we be doing this ourselves?

Reply:
Maybe change this line into an assert or raise Error if not in eval?

Reply:
Oh this is a mistake. Will revert.